src/analyzer/analyzerbeats.cpp

```diff
@@ -136,7 +120,7 @@ bool AnalyzerBeats::shouldAnalyze(TrackPointer pTrack) const {
 
     QString pluginID = m_bpmSettings.getBeatPluginId();
     if (pluginID.isEmpty()) {
-        pluginID = defaultPlugin().id();
+        pluginID = defaultPlugin(availablePlugins()).id();
```

Comment:
Idea: you can make availablePlugins() an abstract virtual function ( = 0 ) of AnalyzerPluginSupportInfo;
Than we get around passing availablePlugins(). 
Do we always use the id()? You may consider to return that directly.